osf/models/mixins.py

```diff
@@ -2092,10 +2103,11 @@ def flag_spam(self):
         """
         super(SpamOverrideMixin, self).flag_spam()
         if settings.SPAM_FLAGGED_MAKE_NODE_PRIVATE:
+            was_public = self.is_public
```

Comment:
Minor: is it theoretically possible to get here in the act of of making something public, but before the object has been updated? I think "no," because this should be hit in the `save`/`on_update` flow